src/debug/ee/controller.cpp

```diff
@@ -6328,7 +6384,19 @@ void DebuggerStepper::TrapStepOut(ControllerStackInfo *info, bool fForceTraditio
         }
         else 
 #endif // FEATURE_MULTICASTSTUB_AS_IL
-        if (info->m_activeFrame.managed)
+        if (info->m_activeFrame.md != nullptr && info->m_activeFrame.md->IsILStub() &&
+            info->m_activeFrame.md->AsDynamicMethodDesc()->GetILStubResolver()->GetStubType() == ILStubResolver::TailCallCallTargetStub)
+        {
+            // Normally the stack trace would not include IL stubs, but we
+            // include this specific IL stub so that we can check if a call into
+            // the tailcall dispatcher will result in any user code being
+            // executed or will return and allow a previous tailcall dispatcher
+            // to deal with the tailcall. Thus we just skip that frame here.
```

Comment:
Lets clarify that info->m_activeFrame.managed = true for these frames, so lacking the special case it would have fallen into the frameless managed code case below.

Reply:
Will do.